docs/manual/lua.md

```diff
@@ -337,7 +339,27 @@ end
 ```
 
 This function is deprecated and scheduled for removal in 6.0,
-use `rpm.execute()` instead.
+use `rpm.spawn()` or `rpm.execute()` instead.
+
+#### spawn({command} [, {actions}])
+
+Spawn, aka execute, an external program. (rpm >= 4.20)
+
+`{command}` is a table consisting of the command and its arguments.
+An optional second table can be used to pass various actions related
+to the command execution, currently supported are:
+
+| Action  | Argument(s) | Description
+|---------|----------------------
+| `stdin` | path        | Redirect standard input to path
+| `stdout`| path        | Redirect standard output to path
+| `stderr`| path        | Redirect standard error to path
+
+Returns the command exit status.
```

Comment:
I don't see this tested.

Reply:
I almost agreed :sweat_smile:  but it is actually tested by this:

```
RPMTEST_CHECK([
runroot_other rpmlua -e "print(rpm.spawn({'cat'}, {stdin='aaa'}))"
],
[0],
[nil    No such file or directory       2.0
],
[])

```

Reply:
What *is* missing is the detailed description of the return code... (shared by rpm.execute())

Reply:
I see:

```lua
> rpm.execute('true')
0,0
> rpm.execute('false')
nil	"exit code"	256,0
```

Reply:
A brief description of the return status added.

Reply:
Hmm, this is strange:

```
> print(rpm.execute('ls blah'))
nil     No such file or directory       2.0
> print(rpm.spawn({'ls', 'blah'}))
ls: cannot access 'blah': No such file or directory
nil     exit code       2
```

Shouldn't the second invocation's error message be the same as that of the first one? IOW, shouldn't we use `WEXITSTATUS(status)` in the error handling code, like we do for `rpm.execute()`?

Reply:
Interesting. The difference comes from executing posix_spawn_file_actions_init(), even if it's empty. And rpm.spawn() always initializes it to keep things simple, under the assumption that an empty set doesn't make any difference, right? :smile: Good spotting, thanks.

Reply:
Except ... nope :smile:  For a moment there I thought I was losing my mind when I couldn't reproduce this in the test-suite, but those two are not equivalent at all. `rpm.execute('ls blah')`  is trying to execute a binary `ls blah` with no arguments, and because there isn't such an executable on most systems, you get a "No such file or directory". The spawn equivalent is `print(rpm.spawn({'ls blah'})` and with that, you get the same result.

Reply:
Oh, damn... Of course, `rpm.execute()` takes a *list* of arguments (much like similar spawn functions in e.g. Python), not a string. My bad :sweat_smile:.